src/graph.cpp

```diff
@@ -186,25 +186,25 @@ inline std::vector<std::vector<node_id_t>> Graph::supernodes_to_merge(std::pair<
   // did end up being able to merge after all
   std::vector<node_id_t> temp_vec;
   for (node_id_t a : new_reps)
-    if (to_merge[a].size() == 0) temp_vec.push_back(a);
+    if (to_merge[a].empty()) temp_vec.push_back(a);
   std::swap(new_reps, temp_vec);
 
   // add to new_reps all the nodes we will merge into
   for (node_id_t a = 0; a < num_nodes; a++)
-    if (to_merge[a].size() != 0) new_reps.push_back(a);
+    if (!to_merge[a].empty()) new_reps.push_back(a);
 
   reps = new_reps;
   return to_merge;
 }
 
 inline void Graph::merge_supernodes(Supernode** copy_supernodes, std::vector<node_id_t> &new_reps,
-               std::vector<std::vector<node_id_t>> &to_merge, bool make_copy, bool first_round) {
+               std::vector<std::vector<node_id_t>> &to_merge, bool first_round, bool make_copy) {
   bool except = false;
   std::exception_ptr err;
   // loop over the to_merge vector and perform supernode merging
   #pragma omp parallel for default(shared)
-  for (node_id_t i = 0; i < new_reps.size(); i++) {
-    node_id_t a = new_reps[i]; // fix for ubuntu18.04 ('for each' style in omp no good)
+  for (node_id_t i = 0; i < new_reps.size(); i++) { // NOLINT(modernize-loop-convert)
```

Comment:
What exactly does this comment mean?

Reply:
It’s a directive for CLion so it doesn’t mark this as a loop that should be modernized

Reply:
Could you add the comment below back in. I want to explain why we aren't using the modern style.